spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMerge.java

```diff
@@ -280,29 +279,30 @@ public void testMergeWithSourceFromSetOps() {
   }
 
   @Test
-  public void testMergeWithMultipleUpdatesForTargetRow() {
+  public void testMergeWithMultipleUpdatesForTargetRowSmallTargetLargeSource() {
```

Comment:
@rdblue, this is the test we talked about. Right now, we are kind of safe as Spark broadcasts the left side of the join only if the join type is inner or right outer. In copy-on-write, it is either full outer or left outer.

Reply:
Looks good to me.